generate/genqlient_directive.go

```diff
@@ -77,7 +118,37 @@ func (dir *genqlientDirective) add(graphQLDirective *ast.Directive, pos *ast.Pos
 		return errorf(pos, "the only valid comment-directive is @genqlient, got %v", graphQLDirective.Name)
 	}
 
+	// First, see if this directive has a "for" option;
+	// if it does, the rest of our work will operate on the
+	// appropriate place in FieldDirectives.
 	var err error
+	forField := ""
+	for _, arg := range graphQLDirective.Arguments {
+		if arg.Name == "for" {
+			err = setString("for", &forField, arg.Value, pos)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	if forField != "" {
+		forParts := strings.Split(forField, ".")
+		if len(forParts) != 2 {
+			return errorf(pos, `for must be of the form "MyType.myField"`)
+		}
+		typeName, fieldName := forParts[0], forParts[1]
+
+		fieldDir := newGenqlientDirective(pos)
+		if dir.FieldDirectives[typeName] == nil {
+			dir.FieldDirectives[typeName] = make(map[string]*genqlientDirective)
+		}
```

Comment:
Can this happen?  I guess it doesn't hurt to be defensive, but if all genqlientDirectives are made via newGenqlientDirective then this `if` will never be true.

Reply:
Yes -- since it's a map of maps we can only initialize the toplevel one upfront.  (It really pains me that Go didn't make the zero value more usable here, but they didn't.)

Reply:
Ah, I see.  Yeah, being forced to call `make` to make maps is pretty annoying.  You don't have to do it for slices!, so...